lib/msf/ui/console/command_dispatcher/creds.rb

```diff
@@ -471,6 +472,11 @@ def creds_search(*args)
           jtr_val = core.private.jtr_format ? core.private.jtr_format : ''
         end
 
+        cracked_hash_origin = Metasploit::Credential::Origin::CrackedPassword.find_by(metasploit_credential_core_id: core.id)
+        cracked_password_core = query.find_by(origin_id: cracked_hash_origin.id, origin_type: "Metasploit::Credential::Origin::CrackedPassword") if cracked_hash_origin.present?
+        cracked_cred_ids << cracked_password_core.id if cracked_password_core.present?
+        cracked_password_val = cracked_password_core.present? ? cracked_password_core.private.data : ''
```

Comment:
One idea might be to enhance the query to group the related cores/origins together and then yield results only when when finished processing all related cores/origins.

I can look into this in more detail, however it will be a couple days. Please take pass at the idea and see what you come up with.

Reply:
I have tried to optimize it using the joins and I have returned the cracked passwords from the `filtered_query` method.